calico.go

```diff
@@ -44,80 +44,91 @@ func init() {
 	// since namespace ops (unshare, setns) are done for a single thread, we
 	// must ensure that the goroutine does not jump from OS thread to thread
 	runtime.LockOSThread()
-
-	nodename, _ = os.Hostname()
-}
-
-func updateNodename(conf NetConf, logger *log.Entry) {
-	if conf.Hostname != "" {
-		nodename = conf.Hostname
-		logger.Warn("Configuration option 'hostname' is deprecated, use 'nodename' instead.")
-	}
-	if conf.Nodename != "" {
-		nodename = conf.Nodename
-	}
 }
 
 func cmdAdd(args *skel.CmdArgs) error {
 	// Unmarshal the network config, and perform validation
-	conf := NetConf{}
+	conf := types.NetConf{}
 	if err := json.Unmarshal(args.StdinData, &conf); err != nil {
 		return fmt.Errorf("failed to load netconf: %v", err)
 	}
 
-	cniVersion := conf.CNIVersion
+	utils.ConfigureLogging(conf.LogLevel)
 
-	ConfigureLogging(conf.LogLevel)
+	// Allow the nodename to be overridden by the network config
+	nodename = utils.DetermineNodename(conf)
 
-	workload, orchestrator, err := GetIdentifiers(args)
+	// Extract WEP identifiers such as pod name, pod namespace (for k8s), containerID, workload name.
+	epIDs, err := utils.GetIdentifiers(args, nodename)
 	if err != nil {
 		return err
 	}
 
-	logger := CreateContextLogger(workload)
-
-	// Allow the nodename to be overridden by the network config
-	updateNodename(conf, logger)
-
-	logger.WithFields(log.Fields{
-		"Orchestrator": orchestrator,
-		"Node":         nodename,
-		"Workload":     workload,
-		"ContainerID":  args.ContainerID,
-	}).Info("Extracted identifiers")
+	logger := &log.Entry{}
```

Comment:
since you are never using that value, it's more common I think to just define the var, so:

`var logger *log.Entry`